Improve exception message

greg0ire opened the pull request:
This was done in reaction to this discussion: https://github.com/doctrine/orm/discussions/9645

In setups where you have many parameters, or do not even realise you are
using an entity, that additional piece of context can be helpful. The
parameter name is not always available where the old exception was
called though.

derrabus reviewed (commented):
Can we add a test for the new exception message?

greg0ire committed:
Improve exception message

In setups where you have many parameters, or do not even realise you are
using an entity, that additional piece of context can be helpful. The
parameter name is not always available where the old exception was
called though.

greg0ire commented on tests/Doctrine/Tests/ORM/Functional/Ticket/DDC2084Test.php:
```diff
@@ -54,10 +54,15 @@ public function testIssue(): void
+Binding entities to query parameters only allowed for entities that have an identifier.
+Class "Doctrine\Tests\ORM\Functional\Ticket\DDC2084\MyEntity2" does not have an identifier.
+EXCEPTION
+        );
```
@derrabus isn't this change in this test good enough?

derrabus replied:
😓 Sorry.
